### PR TITLE
Kai 398 rest api

### PR DESCRIPTION
**Pull request by vita-stejskal**
This add the REST API server for the SQL-Bot. There are several endpoints available:

* `/docs` -- shows Swagger UI
* `/sql-bot/invoke` -- calls the agent with the human input and the chat history; return the AI answer
* `/sql-bot/stream` -- the same as `/sql-bot/invoke` but the server streams the response back as a sequence of "events" that show the steps taken by the agent as well as its final answer. The response stream is a textual output with the [server-sent events](https://en.wikipedia.org/wiki/Server-sent_events)

### CODE REVIEW

File: restapi.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        ],
+        root_path=args.server_path,
+    )
+    langserve.add_routes(
+        app,
+        runnable=_create_api_chain(),
+        enabled_endpoints=['invoke', 'stream']
+    )
+
+    uvicorn.run(app, host=args.bind, port=args.port)
```
Comment (mudrunka) [inline restapi.py:93]: I don't think this is the right way how to start the server.

Instead, we should just create the `app` here - exposing it as a global variable. Then we can start the server by running:
```
gunicorn --bind localhost:5000 \
         --workers 2 \
         --worker-class uvicorn.workers.UvicornWorker \
         'restapi.py:APP'
```

We may still keep this `main` method, but only for test/debug purposes.

**Review by mudrunka — APPROVED** [review]

File: agent.py [hunk 2]
```diff
@@ -37,3 +53,161 @@ def create_agent(self):
             memory=self.memory
         )
         return agent_executor
+
+
+T = TypeVar('T')
```
Comment (Kristyna-Navitas) [inline agent.py:61]: I learned another useful thing, thank you @vita-stejskal  :)

File: agent.py [hunk 3]
```diff
@@ -37,3 +53,161 @@ def create_agent(self):
...
+    def with_langchain_tools(self) -> 'AgentBuilder':
+        self._toolkit = AgentBuilder.Toolkit.LANGCHAIN
+        return self
+
+    def with_toolkit(self, toolkit: 'AgentBuilder.Toolkit') -> 'AgentBuilder':
+        self._toolkit = _check_empty(toolkit, message='toolkit is None')
+        return self
+
+    def with_model(self, model: 'AgentBuilder.Model') -> 'AgentBuilder':
+        self._model = _check_empty(model, message='toolkit is None')
```
Comment (Kristyna-Navitas) [inline agent.py]: model is None, not toolkit

File: restapi.py (hunk 1 above)
Comment (vita-stejskal) [reply restapi.py:93]: Good point, thanks! I've moved the creation of the FastAPI application to a separate function which can be used by `gunicorn`. See 9544a29.

File: agent.py (hunk 3 above)
Comment (vita-stejskal) [reply agent.py]: Fixed in de1a04a, thanks!

**Review by mudrunka — APPROVED** [review]

File: restapi.py (hunk 1 above)
Comment (mudrunka) [reply restapi.py:93]: I think we actually need to create the "static" variable with the initialized app. Like `APP = create_app()` in the `restapi.py`.

**Review by Kristyna-Navitas — APPROVED** [review]

**Pull request merged**